src/slackeros/ros_connector.py

```diff
@@ -95,6 +103,47 @@ def __init__(
         self.queue_worker.setDaemon(True)
         self.queue_worker.start()
 
+    def log_image(self, type=None, topic='/head_xtion/rgb/image_color'):
+        class ImageUploader(Thread):
+
+            def __init__(self, access_token, bridge, send_image, type=None, topic='/head_xtion/rgb/image_color'):
+                Thread.__init__(self)
+                self.access_token = access_token
+                self.bridge = bridge
+                self.send_image = send_image
+                self.type = type
+                self.topic = topic
+
+            def run(self):
+                # upload image
+                image = rospy.wait_for_message(self.topic, Image, timeout=1.5)
+                if self.type is None:
+                    self.type = image.encoding
+                try:
+                    # Convert your ROS Image message to OpenCV2
+                    cv2_img = self.bridge.imgmsg_to_cv2(image, self.type)
+                except CvBridgeError, e:
+                    rospy.logwarn(e)
+                else:
+                    # Save your OpenCV2 image as a jpeg
+                    image_file = '/tmp/camera_image.jpeg'
```

Comment:
I recommend _not_ using a fixed name here, as this could lead to severe race conditions when multiple processes/threads are running. Better use [`mkstemp`](https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp) in python the generate a safe file name. and encapsulate it all in `try...except...finally` where in finally the file is deleted if it exists.